bootstrap/new.py

```diff
@@ -1,64 +1,88 @@
+
+    return content
+
+
+def write_files(files, project_name):
+    for f in files:
+        content = get_file_content(f, project_name)
+        f.write_text(content)
+
+
+def get_files(directory):
+    dir_name = directory.stem
+    if dir_name == "options":
+        return [directory / "abstract.yaml"]
+
+    to_ret = []
+    if dir_name != "models":
+        to_ret.append(directory / "__init__.py")
+
+    to_ret.append(directory / "factory.py")
+    custom_file = f"my{dir_name[:-1]}.py"
+    to_ret.append(directory / custom_file)
+
+    return to_ret
 
 
 if __name__ == "__main__":
     args = parser.parse_args()
-    path = Path(args.project_path)
+    project_name = args.project_name
 
+    path = Path(f"{project_name.lower()}.bootstrap.pytorch")
     path.mkdir()
-    path = path / args.project_name
+    os.mkdir(path / "logs")
+    path = Path(f"{project_name.lower()}.bootstrap.pytorch/{project_name.lower()}")
     path.mkdir()
 
+    print(f"Creating logs directory")
+
+    print("Creating models directory and __init__ file")
     Path(path / "models").mkdir()
     Path(path / "models/__init__.py").touch()
-    Path(path / "models/networks").mkdir()
-    Path(path / "models/networks/__init__.py").touch()
-    Path(path / "models/networks/factory.py").touch()
-    Path(path / "models/networks/mynetwork.py").touch()
-
-    with open(path / "models/networks/factory.py", "w") as f:
-        f.write(r"""from bootstrap.lib.options import Options
-from bootstrap.lib.logger import Logger
-from bootstrap.models.networks.data_parallel import DataParallel
-
-from .mynetwork import MyNetwork
-
-
-def factory(engine):
-    logger = Logger()
-    net_opt = Options()["model"]["network"]
-    logger("Creating Network")
-    if net_opt["name"] == "mynetwork":
-        # You can use any param to create your network
-        # You just have to write them in your option file from options/ folder
-        net = MyNetwork(net_opt["param1"], net_opt["param2"])
-    else:
-        raise ValueError(opt["name"])
-    logger("Network was created")
-
-    if torch.cuda.device_count() > 1:
-        net = DataParallel(net)
-
-    return net
-""")
-
-    with open(path / "models/networks/mynetwork.py", "w") as f:
-        f.write(r"""import torch.nn as nn
-
-
-class MyNetwork(nn.Module):
-    def __init__(self, *args, **kwargs):
-        super(MyNetwork, self).__init__()
-        # Assign args
-
-    def forward(self, x):
-        # x is a dictionnary given by Dataset class
-        pred = self.net(x)
-        return pred  # This is a tensor (or several tensors)
-""")
+
+    directories = [
+        "datasets",
+        "models/networks",
+        "models/criterions",
+        "models/metrics",
+    ]
+
+    for directory in directories:
+        print(f"Creating {directory} folder and associated files")
+        new_dir = path / directory
+        if directory != "models":
+            new_dir.mkdir()
+        files = get_files(new_dir)
```

Comment:
As specified [here](https://github.com/Cadene/bootstrap.pytorch/pull/19#issuecomment-605525218), we would like to follow this path pattern `bootstrap/templates/default/project/datasets/dataset.py` instead of your current `bootstrap/template/datasets/template_dataset.py`.